### ISSUE TITLE
How to decide value of max_iter with respect to our custom dataset?

### ISSUE THREAD

**Issue by dsbyprateekg** [dbolya/yolact#701]
Hi,
Is there any formula to decide max_iter value?
I have total 1300 training +1300 validation images and I have given max_iter values as 55000. Is it right?
Please share your thoughts on it.

**Comment by Elizbellou** [comment]
Hi, don't know if u are still interested, your total dataset is 2600? Then u should split it like 70%-20%-10% (train-valid-test) or 75-15-10 and then start training. If mAP is not satisfactory, I think u should let it for e.g. 100k iteration and monitor behaviour, like how the loss reduces (the model converges), whether mAP is stuck at some point for many iterations etc.. And then tune some hyperparameters in config and check again (learning rate is one of the most important) . Set it lower and then higher an try again. And don't forget to adjust max iterations and learning steps. As long as mAP improves and loss probably reduces as well, you should go further. You can adjust max iterations and resume training from where u stopped.
